packages/client3/components/design-system/sidebar/sidebar.tsx

```diff
@@ -0,0 +1,87 @@
+import clsx from "clsx";
+import { useRouter } from "next/router";
+
+import { FreeArtistsLogoFull } from "../logo";
+import { Toggle, ToggleProps } from "../toggle";
+import { Heading } from "../typography";
+
+interface SideBarProps extends ToggleProps {
+  /**
+   * Sidebar Labels
+   */
+  labels: Array<string>;
+  className?: string;
+  pathName?: string;
+  storyBookMode?: string;
+}
+
+export function SideBar({
+  labels,
+  className,
+  states,
+  pathName,
+  storyBookMode,
+}: SideBarProps) {
+  const router = useRouter();
+  const handleClick = (event: any, href: string, pathName?: string) => {
+    if (pathName) {
+      return;
+    } else {
+      event.preventDefault();
+      router.push(href);
+    }
+  };
+
+  return (
+    <div
+      className={clsx("absolute h-full w-72 bg-green-90 shadow-md", className)}
+      id="sideBar"
+    >
+      <div className="px-6 pt-4 pb-9">
+        <a href="#!">
+          <div className="flex justify-center">
+            <div className="shrink-0">
+              <FreeArtistsLogoFull storyBookMode={storyBookMode} />
+            </div>
+          </div>
+        </a>
+      </div>
+      <div className="flex shrink-0 justify-center pb-10">
+        <Toggle states={states} />
+      </div>
+      <ul className="relative ">
+        {labels.map((label: string) => (
+          <li key={label} className="relative">
+            <a
+              href={pathName ? "" : label}
+              onClick={
+                pathName
+                  ? (event) => handleClick(event, label, pathName)
```

Comment:
Is this function call necessary, as the function seems to exit early if pathname is present?